metabase/metabase
# Enable hiding/showing individual events in visualizations

Solves #26869

### How to Test

Please check with visual and native questions.

a. Example of a visual question: `Sample Database › Orders, Summarize using Count by Created at: Month`
b. Example of a native query: `SELECT ID, CREATED_AT FROM ORDERS limit 1000`

Visualize question using Bars and Lines, among others

1. Click on the calendar icon on the bottom right
2. Add timeline events
3. Click on the checkbox next to each timeline event to hide/show it in the viz

#### Highlighted events

Events can be highlighted by clicking on their divs in the sidebar.

If hidden, atm an event cannot be highlighted.

If hidden when already highlighted, once made visible again, the event will be highlighted.

Highlighting a hidden event will no longer make a parent timeline visible in case it was hidden.

🎗️ I'm waiting on a look from @kdoh and @cdeweyx on the above.

Demo:
https://www.loom.com/share/5e2f5747e45b4ea0a54646c580f9d453

#### Timelines that are hidden on page load

On page load, timelines from collections other than the collection a question is in are hidden.
This is unchanged in this PR.
All child events of these hidden events will be shown as visible as the user makes their timelines visible.

Demo:
https://www.loom.com/share/724f1ad483be4d468220e7726cec64c3
